### PR TITLE
added pad operator

### PR DESCRIPTION
**Pull request by kshitij01042002**
### Details:
Added `microsoft.com.pad` operator.

### Tickets:
 - Closes https://github.com/openvinotoolkit/openvino/issues/17576

### CODE REVIEW

**Comment by kshitij01042002** [comment]
Hi @p-wysocki @gkrivor can you PTAL at the PR. I have created a draft he the PR looks good I'll proceed to adding the test cases for the same.

File: src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+OutputVector pad(const Node& node);
+}  // namespace set_1
+}  // namespace op
+}  // namespace onnx_import
+}  // namespace ngraph
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: could you please add a test using a test model like there https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models ?

Comment (kshitij01042002) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: Actually I am not sure after I have added the test model how can I run the tests locally?

Comment (ilya-lavrenov) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: Test models are automatically converted to .onnx files during build of OpenVINO tests (make sure that appropriate dependencies are installed; cmake will notify about it if something is missed, just read cmake output).

For example, we have a model file https://github.com/openvinotoolkit/openvino/blob/749ed9dec7f2817fb3f27838ea3d307bf14c9ebf/src/core/tests/models/ir/split_equal_parts_2d.prototxt#L15, which is converted to `onnx/split_equal_parts_2d.onnx` and used in this test:
https://github.com/openvinotoolkit/openvino/blob/749ed9dec7f2817fb3f27838ea3d307bf14c9ebf/src/frontends/onnx/tests/onnx_import_reshape.in.cpp#L487-L498
To run the tests, go to `<source dir>/bin/intel64/Releases` and run `ov_onnx_frontend_tests` binary

Comment (kshitij01042002) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: Do I need to buld OpenVino before running the tests?

Comment (kshitij01042002) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: FYI: I have just forked cloned the repo and added the code. I'm unsure if the code changes are correct or wrong as I was not able to test it.

Comment (ilya-lavrenov) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: sure, OpenVINO tests are built together with OpenVINO if you enable `-DENABLE_TESTS` in cmake

Comment (kshitij01042002) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: Should I follow these steps https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build_windows.md? I'm setting it up on a windows machine

If I follow these steps the build creates new files, which identifies as untracked files by git, what should I do in this case?

Comment (kshitij01042002) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: Error while running `cmake --build . --config Release --verbose -j8` with flag `-DENABLE_TESTS=ON`.

Error trace:
Unknown argument -DENABLE_TESTS=ON
Usage: cmake --build <dir>             [options] [-- [native-options]]
       cmake --build --preset <preset> [options] [-- [native-options]]
       
 Could you help me with this @p-wysocki @gkrivor @ilya-lavrenov. Thanks.

Comment (ilya-lavrenov) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: Please, follow build guides first https://github.com/openvinotoolkit/openvino/blob/master/docs/dev/build.md

Comment (p-wysocki) [reply src/frontends/onnx/frontend/src/op/com.microsoft/pad.hpp:20]: @kshitij01042002 Here's a CMake command I'm usually using, it should work fine:

```bash
cmake .. -DENABLE_TESTS=ON -DNGRAPH_UNIT_TEST_ENABLE=ON -DENABLE_FUNCTIONAL_TESTS=ON -DENABLE_OPENCV=ON -DENABLE_INTEL_CPU=ON -DENABLE_INTEL_GNA=OFF -DENABLE_INTEL_GPU=OFF -DENABLE_PYTHON=ON -DENABLE_NCC_STYLE=OFF -DCMAKE_INSTALL_PREFIX=/home/pwysocki/openvino_dist -DENABLE_DEBUG_CAPS=ON -DENABLE_CPU_DEBUG_CAPS=ON -DPYTHON_EXECUTABLE=`which python`
```
Please let mi know if it works for you. After that I usually just ran `make -jX install`.

**Pull request closed without merging**